xmsannotator/R/compute_scores.R

```diff
@@ -1,35 +1,102 @@
+#' Assign abundance ratio of the second most abundant isotope/isotopologue for each entry in a table
+#'
+#' @param isotopes A table with annotated peaks filtered by adducts
+#' @return The input table with extra column containing the abundance ratio of the second most abundant isotope/isotopologue
+#'
+#' @import dplyr
+#' @import purrr
+assign_isotope_abundances <- function(isotopes) {
+  molecules <- distinct(isotopes, molecular_formula)
+  molecules <- mutate(molecules, abundance_ratio = map_dbl(molecular_formula, compute_abundance_ratio))
+  isotopes <- inner_join(isotopes, molecules, by = "molecular_formula")
+}
+
+#' Get abundance ratio of the second most abundant isotope/isotopologue
+#'
+#' @param formula A string containing molecular or empirical formula of a compound
+#'
+#' @return Relative abundance ratio of the second most abundant isotope/isotopologue
+#'
+#' @importFrom Rdisop getMolecule
+compute_abundance_ratio <- function(formula) {
+  molecule <- getMolecule(formula)
+  abundance_ratios <- sort(molecule$isotopes[[1]][2,], decreasing = TRUE)
+  sec_most_abundant <- abundance_ratios[2]
+}

```

Comment:
So that means we are only checking if the next most abundant isotopologue fits?

Reply:
We use the intensity of the second most abundant isotopologue as a cutoff, below which other isotopes may lay. As we discussed, this is a prefiltering, and we will check if the intensities of all identified isotopes fit their actual abundances in a later step.